serving/purgatorio/select-and-train-machine-learning-models/tools-and-libraries/tools-and-libraries.md

```diff
@@ -1,5 +1,321 @@
+
+We'll give you some general tips to learn effectively and develop rock-solid foundations, that you can rely on to address and solve Data Science problems in the complexity of the real world (which is messy by definition).
+
+In this guide, you'll find a ton of different small tools, big libraries, and even entire frameworks, but be aware!
+
+As stated in the [Virgilio's Teaching Strategy](https://github.com/virgili0/Virgilio/blob/master/serving/paradiso/virgilio-teaching-strategy/virgilio-teaching-strategy.md) Guide, the best way to deal with the learning process is to leverage the "[Divide et Impera](https://en.wikipedia.org/wiki/Divide_and_rule)" principle, which states that every time you're tackling a problem (in this case, the _learning problem_) you should split it into smaller parts, easier to handle and understand. 
+
+# Solving the Learning Problem 
+
+The process of learning a tool and use it effectively **is equal to solving a problem**.
+
+We consider "solved" the problem when you can use the tool comfortably.
+
+So, you shouldn't try to learn all the features of a library in the same run, or you're going to be overwhelmed! 
+
+Every rich Python library contains thousands of different functions and utilities, but _you'll need a small subset to get the things done!_
+
+**So, Virgilio suggests the following approach**:
+
+- Choose a library from the ones that are widely used and feature-rich.
+- Get a general introduction through to Virgilio's resources listed in this guide.
+- Find a problem (similar but not equal to the introductory ones).
+- Dig into the problem, and get your hands dirty with the chosen library.
+- Learn _on the way to the problem solution_.
+- Solve the problem, trying to do it yourself.
+
+If you follow this approach, you take home the following 3 things:
+- **real problem experience**
+- **you feel comfortable coding with a subset of the functions of the tool**
+
+Believe it or not, it works like magic!
+
+Know you know how to solve a new _learning problem_.
+
+Now, how to choose from the myriad of tools and libraries available?
+
+# Navigating the Ocean
+
+It's very easy to find out articles, blog posts, and YouTube videos telling you "the 20 libraries you need to learn in 2020!". 
+
+What? 20 libraries? 
+
+Come on, this is ridiculous _and_ impossible, for a beginner!
+
+Don't even try to do that, or you will miserably get stuck, because it is impossible to learn 20 programming libraries together, even for an expert.
+
+Learning tools and libraries will be a thing most of your life if you want to study or work in the Data Science field. 
+
+**But, as a beginner**:
+
+- is better to start learning very well _some_ tools/libraries, and meanwhile becoming "aware" of the similar ones (and their pro/cons).
+
+- For doing that, it's useful to read the homepage of the official documentation (God bless good documentation!). 
```

Comment:
Good advise but this isn't very common, it can get outdated fast, as the tool moves really quickly.

Reply:
yes, but somewhere one has to start :D